Utilities/Mergers/test/histosTopologyCommon.h

```diff
@@ -0,0 +1,139 @@
+      specs.push_back(DataProcessorSpec{
+        std::string{"producer-histo"} + std::to_string(producerIdx),
+        Inputs{},
+        Outputs{{{"mo"}, origin, description, static_cast<SubSpecificationType>(producerIdx), Lifetime::Sporadic}},
+        AlgorithmSpec{
+          static_cast<AlgorithmSpec::ProcessCallback>([histoBinsCount = mHistoBinsCount, histoMin = mHistoMin, histoMax = mHistoMax, producerIdx, dataSent = false](ProcessingContext& processingContext) mutable {
+            if (dataSent) {
+              std::this_thread::sleep_for(std::chrono::milliseconds{100});
+              return;
+            }
+            TH1F& histo = processingContext.outputs().make<TH1F>(
+              Output{origin, description, static_cast<SubSpecificationType>(producerIdx)},
+              "histo", "histo", histoBinsCount, histoMin, histoMax);
+            histo.Fill(5);
+            histo.Fill(producerIdx);
+            dataSent = true;
+          })}});
+    }
+    return inputs;
+  }
+
+  void generateMergers(WorkflowSpec& specs, const Inputs& producerInputs, mergers::InputObjectsTimespan mergerType)
+  {
+    using namespace mergers;
+
+    MergerInfrastructureBuilder mergersBuilder;
+    mergersBuilder.setInfrastructureName("histos");
+    mergersBuilder.setInputSpecs(producerInputs);
+    mergersBuilder.setOutputSpec({{"main"}, origin, description, 0});
+    MergerConfig config;
+    config.inputObjectTimespan = {mergerType};
+    std::vector<std::pair<size_t, size_t>> param = {{5, 1}};
+    config.publicationDecision = {PublicationDecision::EachNSeconds, param};
+    config.mergedObjectTimespan = {MergedObjectTimespan::FullHistory};
+    config.topologySize = {TopologySize::NumberOfLayers, 2};
+    mergersBuilder.setConfig(config);
+
+    mergersBuilder.generateInfrastructure(specs);
+  }
+
+  void generateChecker(WorkflowSpec& specs)
+  {
+    specs.push_back(DataProcessorSpec{
+      "data-checker",
+      Inputs{{"histo", origin, description, 0, Lifetime::Sporadic}},
+      Outputs{},
+      AlgorithmSpec{
+        AlgorithmSpec::InitCallback{[expectedResult = mExpectedResult](InitContext&) {
+          // reason for this crude retry is that multiple layers are not synchronized between each other and publish on their own timers
+          return AlgorithmSpec::ProcessCallback{[expectedResult, retryNumber = 1, retries = 5](ProcessingContext& processingContext) mutable {
+            const auto histo = processingContext.inputs().get<TH1F*>("histo");
+            if (HistosMergerTestGenerator<HistoSize>::compareHistoToExpected(expectedResult, *histo.get())) {
+              processingContext.services().get<ControlService>().readyToQuit(QuitRequest::All);
+              return;
+            } else {
+              if (retryNumber++ == retries) {
+                processingContext.services().get<ControlService>().readyToQuit(QuitRequest::All);
+                LOG(fatal) << "received incorrect data";
+              }
+            }
```

Comment:
```suggestion
            } else if (retryNumber++ == retries) {
                processingContext.services().get<ControlService>().readyToQuit(QuitRequest::All);
                LOG(fatal) << "received incorrect data";
            }
```
any reason not to simplify it?

Reply:
good catch after n iterations this is what I was left with and copied it around